### PR TITLE
Upgrade AWS GoSDK version to v1.38.11

### PR DESCRIPTION
**Pull request by akartsky**
Upgrade to the latest version of AWS GoSDK.

This update includes the Artifactory Auth Support for Hosting 

All integ tests passed

### CODE REVIEW

**Review by RedbackThomson — APPROVED** [review]
Oh great, this also includes environment variable support for training!

File: config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml [hunk 1]
```diff
@@ -17,12 +17,12 @@ spec:
...
   - additionalPrinterColumns:
-    - jsonPath: .status.failureReason
-      name: FailureReason
-      type: string
     - jsonPath: .status.hyperParameterTuningJobStatus
       name: HyperParameterTuningJobStatus
       type: string
+    - jsonPath: .status.failureReason
+      name: FailureReason
+      type: string
```
Comment (jaypipes) [inline config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml:25]: @RedbackThomson didn't we have a fix for this non-deterministic sorting dilemma?

Comment (RedbackThomson) [reply config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml:25]: I never worked on one, no. I looked into it based on your suggestion but it was non-trivial to me and I guess it was never prioritised.

Comment (mbaijal) [reply config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml:25]: Create a github issue @RedbackThomson if there isnt one ?

**Review by mbaijal — APPROVED** [review]
Did you sync with Suraj's changes as discussed ?

File: config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml (hunk 1 above)
Comment (RedbackThomson) [reply config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml:25]: https://github.com/aws-controllers-k8s/community/issues/720 Opened 14 days ago

Comment (surajkota) [reply config/crd/bases/sagemaker.services.k8s.aws_hyperparametertuningjobs.yaml:25]: @akartsky please git add only changes unrelated to this

**Review by surajkota — CHANGES_REQUESTED** [review]
Do NOT merge this PR as-is. Please rebase once #7 is merged and use the latest version(`338d010225939a25b2e175594fec29b56707b388`) of code generator and runtime 0.0.5

Also please git add files necessary for the PR

**Comment by akartsky** [comment]
> Do NOT merge this PR as-is. Please rebase once #7 is merged and use the latest version(`338d010225939a25b2e175594fec29b56707b388`) of code generator and runtime 0.0.5
> 
> Also please git add files necessary for the PR

Made all changes 
ready to merge

File: apis/v1alpha1/endpoint.go [hunk 2]
```diff
@@ -90,8 +90,8 @@ type EndpointStatus struct {
 // Endpoint is the Schema for the Endpoints API
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
-// +kubebuilder:printcolumn:name="FailureReason",type=string,JSONPath=`.status.failureReason`
```
Comment (surajkota) [inline apis/v1alpha1/endpoint.go:93]: nit: revert

**Review by surajkota — CHANGES_REQUESTED** [review]

**Review by surajkota — APPROVED** [review]

**Pull request merged**